src/primitives.rs

```diff
@@ -29,7 +29,7 @@ pub(crate) fn get_block_env(header: &Header) -> Option<BlockEnv> {
 #[derive(Debug, Clone, PartialEq)]
 pub enum TransactionParsingError {
     OverflowedGasLimit,
-    MissingGasPrice,
+    GetGasPriceError(String),
```

Comment:
We can parameterize `TransactionParsingError ` by `PevmChain` to reuse its error type?